edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/ServiceProxy.cs

```diff
@@ -94,24 +98,31 @@ public async Task<Option<ServiceIdentity>> GetServiceIdentity(string deviceId)
             return serviceIdentityResult;
         }
 
-        public async Task<Option<ServiceIdentity>> GetServiceIdentity(string deviceId, string moduleId)
+        public async Task<Option<ServiceIdentity>> GetServiceIdentity(string deviceId, string moduleId, string onBehalfOfDevice)
         {
+            Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
+            Preconditions.CheckNonWhiteSpace(moduleId, nameof(moduleId));
+            Preconditions.CheckNonWhiteSpace(onBehalfOfDevice, nameof(onBehalfOfDevice));
+
             string id = $"{deviceId}/{moduleId}";
             Option<ScopeResult> scopeResult = Option.None<ScopeResult>();
 
             try
             {
                 IDeviceScopeApiClient client;
+                ScopeResult res;
+
                 if (this.nestedEdgeEnabled)
                 {
                     client = this.securityScopesApiClientProvider.CreateNestedDeviceScopeClient();
+                    res = await client.GetIdentityOnBehalfOfAsync(deviceId, Option.Some(moduleId), onBehalfOfDevice);
                 }
                 else
                 {
                     client = this.securityScopesApiClientProvider.CreateDeviceScopeClient();
+                    res = await client.GetIdentityAsync(deviceId, moduleId);
                 }
 
-                ScopeResult res = await client.GetIdentityAsync(deviceId, moduleId);
                 scopeResult = Option.Maybe(res);
```

Comment:
Even you can remove "ScopeResult res" variable and directly assign to scopeResult variable.
```suggestion
                scopeResult = Option.Some(res);
```